src/alchemlyb/__init__.py

```diff
@@ -46,15 +46,6 @@ def concat(objs, *args, **kwargs):
  
  
     .. versionadded:: 0.5.0'''
-    # Sanity check
-    try:
-        attrs = objs[0].attrs
-    except IndexError: # except empty list as input
-        raise ValueError('No objects to concatenate')
-
-    for obj in objs:
-        if attrs != obj.attrs:
-            raise ValueError('All pandas objects should have the same attrs.')
-    new = pd.concat(objs, *args, **kwargs)
-    new.attrs = attrs
-    return new
```

Comment:
keep the original code, don't replace with pd.concat

Reply:
(In another discussion on MDAnalysis, @IAlibay convinced me that we should just preserve the deprecated code exactly as is; also minimizes the diff)